pkg/interface/src/views/apps/graph/app.js

```diff
@@ -2,58 +2,53 @@ import React, { PureComponent } from 'react';
 import { Switch, Route } from 'react-router-dom';
 import { Center, Text } from "@tlon/indigo-react";
 import { deSig } from '~/logic/lib/util';
-
-
-export default class GraphApp extends PureComponent {
-  render() {
-    const { props } = this;
-    const contacts = props.contacts ? props.contacts : {};
-    const groups = props.groups ? props.groups : {};
-    const associations =
-      props.associations ? props.associations : { graph: {}, contacts: {} };
-    const graphKeys = props.graphKeys || new Set([]);
-    const graphs = props.graphs || {};
-
-    const { api } = this.props;
-
-    return (
-      <Switch>
-        <Route exact path="/~graph/join/ship/:ship/:name/:module?"
-          render={ (props) => {
-            const resource =
-              `${deSig(props.match.params.ship)}/${props.match.params.name}`;
-            const { ship, name } = props.match.params;
-            const path = `/ship/~${deSig(ship)}/${name}`;
-            const association = associations.graph[path];
-
-
-            const autoJoin = () => {
-              try {
-                api.graph.joinGraph(
-                  `~${deSig(props.match.params.ship)}`,
-                  props.match.params.name
-                );
-                
-                
-              } catch(err) {
-                setTimeout(autoJoin, 2000);
-              }
-            };
-
-            if(!graphKeys.has(resource)) {
-              autoJoin();
-            } else if(!!association) {
-              props.history.push(`/~landscape/home/resource/${association.metadata.module}${path}`);
+import useGraphState from '~/logic/state/graph';
+import useMetadataState from '~/logic/state/metadata';
+
+const GraphApp = (props) => {
+  const associations= useMetadataState(state => state.associations);
+  const graphKeys = useGraphState(state => state.graphKeys);
+
+  const { api } = this.props;
```

Comment:
hm??

Reply:
hm hm? All of these are used in the file.